Backend/BackendAPI/Repository/Repositories/BikeRepository.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using BackendAPI.Data;
+using BackendAPI.Models;
+using BackendAPI.Repository.Interfaces;
+
+namespace BackendAPI.Repository.Repositories
+{
+    public class BikeRepository : GenericRepository<Bike>, IBikeRepository
+    {
+        public BikeRepository(DataContext dbContext) : base(dbContext)
+        { }
+
+        public new bool Delete(int ID)
+        {
+            Bike bike = GetByID(ID);
+            if (bike == null)
+                return false;
+            dbContext.Bikes.Remove(bike);
+            return true;
+
+        }
+
+        public new IList<Bike> Get()
+        {
+            return dbContext.Bikes.ToList();
+        }
+
+        public new Bike GetByID(int ID)
+        {
+            try
+            {
+                return dbContext.Bikes.First(b => b.ID == ID);
+            }
+            catch
+            {
+                return null;
+            }
```

Comment:
Mam wrażenie że lepiej tutaj zrobić FirstOrDefault. Robienie tego wyjątkami może nam potem uniemożliwić wykrycie innych ważnych wyjątków (np. wyskoczy jakiś wyjątek niepowiązany z tym, że nie ma takiego id i ponieważ jest zrobiona obsługa wyjątku to nawet nie będziemy o tym wiedzieć)